py/src/braintrust/logger.py

```diff
@@ -362,11 +362,9 @@ def log(self, *args):
                 self.queue.put_nowait(event)
             except queue.Full:
                 # Notify consumers to start draining the queue.
-                with self.queue_filled_condition:
-                    self.queue_filled_condition.notify()
+                self.queue_filled_semaphore.release()
```

Comment:
I'm not sure I follow the logic here. Once you call `release()` a consumer thread _can_ wake up, but it's not guaranteed that (a) you'll context switch to it and (b) that you'll only context switch here _after_ it has pulled something off the queue. Therefore, I think you can still lose an event here.

In traditional concurrency land, you'd have a condition variable here that's "paired" with a mutex, and the mutex is _released_ when you call `wait`. Then, the thread that dequeues would, after clearing up the queue, acquire hte mutex and notify the condition variable, allowing this thread to proceed.

Reply:
From offline discussion, we shouldn't be at risk of losing data because the subsequent `put` is blocking until the queue has space.